### PR TITLE
Fix bugs add more test cases

### PR DESCRIPTION
**Pull request by JakeHartnell**
Fixes a few bugs:
- Creating > 6 trait proposals lead to index out of bounds error
- `endAuction` doesn't cover tie scenario

Maybe some others as well.

Will explain this because it's a little counter intuitive. Two-dimensional arrays in solidity do not work as you would expect in solidity. :joy: 

See this issue for an explanation of what was wrong with the out of bounds error: https://ethereum.stackexchange.com/questions/11870/create-a-two-dimensional-array-in-solidity

Basically when you make `uint256[8][10]`, setting a variable is the reverse of this. :exploding_head:

### CODE REVIEW

**Review by yaoe — APPROVED** [review]
thanks ! good job spotting the bug ;)
